packages/svelte/src/compiler/utils/ast.js

```diff
@@ -54,41 +54,51 @@ export function is_event_attribute(attribute) {
 }
 
 /**
- * Extracts all identifiers from a pattern.
+ * Extracts all identifiers from a pattern. By default only those which can be newly declared as part of an assignment expression are included.
  * @param {ESTree.Pattern} param
  * @param {ESTree.Identifier[]} [nodes]
+ * @param {boolean} [include_member_expressions] If `true`, will also include member expressions which can be important in a case like `[a, b] = [c, d]`
  * @returns {ESTree.Identifier[]}
  */
-export function extract_identifiers(param, nodes = []) {
+export function extract_identifiers(param, nodes = [], include_member_expressions = false) {
 	switch (param.type) {
 		case 'Identifier':
 			nodes.push(param);
 			break;
 
+		case 'MemberExpression':
+			if (
+				include_member_expressions &&
+				(param.object.type === 'Identifier' || param.object.type === 'MemberExpression')
```

Comment:
Do we need to check if it's computed too?

Reply:
No. In an expression like `[name.a[x].d] = ...;` the inner-most object of the member expression is `name`, which is the variable we're interested in.
Having written that, we can make this code clearer by reflecting that - changing this.